File: iru/week1/medium.js
function solution(clothes) {
  let obj = {};

  for (let i = 0; i < clothes.length; i++) {
    let type = clothes[i][1];

    if (obj[type] === undefined) {
      obj[type] = 1;
    } else {
      obj[type]++;
    }
  }

  let total = 1;
  let counts = Object.values(obj);

  for (let j = 0; j < counts.length; j++) {
    let count = counts[j];
    total *= count + 1;
  }

  return total - 1;
}

File: jimin/week1/medium.py
def solution(clothes):
    counts = {}
    for name, kind in clothes:
        counts[kind] = counts.get(kind, 0) + 1

    ans = 1
    for kind in counts:
        ans *= (counts[kind] + 1)  # 그 종류에서 (입는 경우 수 + 안 입는 경우 1)

    return ans - 1  # 아무것도 안 입는 경우 제외
